api/RequestTypes.d.ts

```diff
@@ -2464,21 +2464,106 @@ export interface WatcherPutWatch<T = RequestBody> extends Generic {
   body?: T;
 }
 
-export interface WatcherStart extends Generic {
+interface WatcherStartRequest extends GenericRequest {
 }
 
-export interface WatcherStats extends Generic {
+interface WatcherStatsRequest extends GenericRequest {
   metric?: string | string[];
   emit_stacktraces?: boolean;
 }
 
-export interface WatcherStop extends Generic {
+interface WatcherStopRequest extends GenericRequest {
 }
 
-export interface XpackInfo extends Generic {
+interface XpackInfoRequest extends GenericRequest {
   categories?: string | string[];
 }
 
-export interface XpackUsage extends Generic {
-  master_timeout?: string;
-}
+interface XpackUsageRequest extends GenericRequest {
+  master_timeout?: string;
+}
+
+export {
```

Comment:
What is the rationale for using this export style?

Reply:
As you can see in the `index.d.ts`, I'm exporting all the request/response type definitions with
```ts
export * from './api/RequestTypes'
export * from './api/ResponseTypes'
```
For some reason that I cannot fully understand, if  use the export style you are suggesting, I was able to also use interfaces that were not exported.
For example:
```ts
/* ResponseTypes.d.ts */
export interface Foo {}
interface Bar {}

/* index.d.ts */
export * from './api/ResponseTypes.d.ts'
export { Client }

/* code.ts */
// here I should be able to import Bar, but I can
import { Client, Foo, Bar } from './'
```

Then I discovered that using the "single export" statement solves this quirk.

Reply:
Okay, this is indeed a quick of `.d.ts.` files, [all their symbols are exported by default](https://github.com/Microsoft/TypeScript/issues/19545). However, I think there should not be any private interfaces in `.d.ts` files, users should be able to access any interface that the can potentially use in their code, e.g. why not exporting `interface Shards`?
The user might want to write the following code:
```ts
const res = await client.update(...);
const shards: Shards = res._shards;
processShards(shards);

// Why not giving the possibility to use the Shards type?
function processShards(shards: Shards) {}
```

Reply:
As you know, the API surface of Elasticsearch is massive, and given that we are doing this work manually, it's very easy to miss something.
There might be a `ShardsResponse` that is different from `Shards` (I don't think this is the case, but it might happen), and I would love to avoid doing too many breaking changes because I wasn't cautious about what I'm exposing.
Adding these response types will be a very long work that will be done during time, and a big part of it will be fix and improve the types already written.
Thank you for your suggestions!